applications/OptimizationApplication/tests/test_model_part_controllers.py

```diff
@@ -0,0 +1,31 @@
+
+import KratosMultiphysics as Kratos
+import KratosMultiphysics.KratosUnittest as kratos_unittest
+import KratosMultiphysics.StructuralMechanicsApplication
+from KratosMultiphysics.OptimizationApplication.utilities.optimization_info import OptimizationInfo
+from KratosMultiphysics.OptimizationApplication.model_part_controllers.mdpa_model_part_controller import MdpaModelPartController
+
+@kratos_unittest.skipIfApplicationsNotAvailable("StructuralMechanicsApplication")
+class TestMdpaModelPartController(kratos_unittest.TestCase):
+    def test_MdpaModelPartController(self):
+        with kratos_unittest.WorkFolderScope("linear_element_test", __file__):
+            model_part_controller_settings = Kratos.Parameters("""{
+                "model_part_name": "Structure",
+                "input_filename" : "Structure",
+                "domain_size"    : 3
```

Comment:
Can you add a test to check both scenarios: read_data: true and false?

Reply:
done :)